src/tools/rbd/action/Snap.cc

```diff
@@ -412,6 +418,14 @@ int execute_remove(const po::variables_map &vm,
     std::cerr << "rbd: trying to access image using both name and id. "
               << std::endl;
     return -EINVAL;
+  } else if (!snap_name.empty() && snap_id != CEPH_NOSNAP) {
```

Comment:
@dillaman this should be below, after `utils::get_pool_image_snapshot_names` or `utils::get_pool_snapshot_names` calls.

Reply:
Tweaked